src/components/Tag/Tag.tsx

```diff
@@ -13,14 +13,28 @@ export interface TagProps {
   children?: string;
   /** Disables the tag  */
   disabled?: boolean;
-  /** Callback when tag is removed */
+  /** Callback when tag is clicked and no remove button is rendered */
+  onClick?(): void;
+  /** Callback when remove button is clicked or keypressed */
   onRemove?(): void;
 }
 
```

Comment:
Very nice - much cleaner suggestion thank you

I feel that it works very well here since we're defining the render of a `<Tag />` depending on it's prop availabilities, and not the value of of one of those props where we'd potentially get different values on the same prop type.